onnxmltools/convert/common/_apply_operation.py

```diff
@@ -162,28 +159,80 @@ def apply_concat(scope, input_names, output_name, container, operator_name=None,
 
     container.add_node('Concat', input_names, output_name, op_version=op_version, name=name, axis=axis)
 
+def apply_constant(scope, input_names=[], output_name, operator_name=None, value=0.0):
+    name = _create_name_or_use_existing_one(scope, 'Constant', operator_name)
+    attrs = {'name': name, 'value': value}
 
-def apply_clip(scope, input_name, output_name, container, operator_name=None, max=None, min=None):
-    name = _create_name_or_use_existing_one(scope, 'Clip', operator_name)
+    if container.target_opset < 9:
```

Comment:
well, if we don't use any attr or schema from opset 9, should we just keep it as 1

Reply:
In opset 9, the constant op was expanded to work with all input/output types. In opset 1, constant only worked with float tensor input. Does it make sense to add a data type check in the opset 1 case?